docs/src/userguide/memoryviews.rst

```diff
@@ -691,6 +691,49 @@ reject None input straight away in the signature, which is supported in Cython
 Unlike object attributes of extension classes, memoryview slices are not
 initialized to None.
 
+
+Pass data from a C function via pointer
+=======================================
+
+Since use of pointers in C is ubiquitous, here we give a quick example of how
+to call C functions whose arguments contain pointers. Let's suppose you want to
+manage an array (allocate and deallocate) with NumPy (it can also be Python arrays, or
+anything that supports the buffer interface), but you want to perform computation on this
+array with an external C function implemented in :file:`C_func_file.c`:
+
+.. literalinclude:: ../../examples/memoryviews/C_func_file.c
+    :linenos:
+
+This file have a header file called :file:`C_func_file.h` containing::
+
+    void multiply_by_10_in_C(double arr[], unsigned int n);
+
+where ``arr`` points to the array and ``n`` is its size.
+
+You can call the function in a Cython file in the following way:
+
+.. literalinclude:: ../../examples/memoryviews/memview_to_c.pyx
+    :linenos:
+
+Several things to note:
+
+ - ``&arr_memview[0]`` can be understood as 'the adress of the first element of the
+   memoryview'.
+ - ``arr_memview.shape[0]`` could have been replaced by ``arr_memview.size``,
+   ``arr.shape[0]`` or ``arr.size``. But ``arr_memview.shape[0]`` is more efficient
+   because it doesn't require a conversion from a Python ``int`` to an
+   unsigned C integer.
+ - ``multiply_by_10`` will perform computation in-place if the array passed is contiguous,
+   and will return a new numpy array if ``arr`` is not contiguous.
+ - If you are using Python arrays instead of numpy arrays, you don't need to check
+   if the data is stored contiguously as this is always the case. See :ref:`array-array`.
+
+This way, you can have access the function more or less as a regular
+Python function while its data and associated memory gracefully managed
```

Comment:
Not entirely sure I understand this sentence. Is it trying to say something like this?
"This way, you can call the C function similar to a normal Python function, and leave all the memory management and cleanup to NumPy arrays and Python's object handling."